### DISCUSSION TITLE
SyntaxError: Unexpected token import on webpack build

### DISCUSSION THREAD

**Discussion by kahibalo01** [webpack/webpack#18736] [Q&A]
Hello, I'm not sure if it is okay to ask here, I've looked everywhere and cannot find the answer.

Running `webpack --mode production`  gives me no errors on my machine.
But on a colleague's machine, it spits out this error:

```
<omitted-path>/node_modules/webpack/bin/webpack.js:82
		import(path.resolve(path.dirname(pkgPath), pkg.bin[cli.binName])).catch(
		^^^^^^

SyntaxError: Unexpected token import
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:617:28)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
```

node version: 16.16.0
npm version:  8.11.0

package.json:
```
{
  "name": "azaark",
  "version": "0.0.1",
  "description": "description",
  "scripts": {
    "build": "webpack --mode production"
  },
  "author": "",
  "license": "ISC",
  "engines": {
    "node": "^16.20.0"
  },
  "devDependencies": {
    "css-loader": "^7.1.2",
    "mini-css-extract-plugin": "^2.9.1",
    "sass": "^1.77.8",
    "sass-loader": "^16.0.1",
    "ts-loader": "^9.5.1",
    "typescript": "^5.5.4",
    "webpack": "^5.94.0",
    "webpack-cli": "^5.1.4"
  }
}
```

We've checked if we have the same node and npm versions, and reinstalled those using nvm, cleaned npm cache, node_modules, and package-lock, but to no avail. We've also tried `npm install` from a fresh project, nada.

It would really help if you can point me to the right direction. I'm pretty lost here.

**Comment by alexander-akait** [comment]
please run `node -v`

**Reply by kahibalo01** [reply]
v16.16.0
This is the version on both machines: the one working and the one that gets this error.

**Reply by kahibalo01** [reply]
It seems the dynamic "import()" is not recognized for some reason.

In fact, I'm not even sure why it is entering that block. Tracing the code, the pkg.type and pkg.bin conditions should have been false on the environment. This is the conditional on webpack.js:

```
    if (pkg.type === "module" || /\.mjs/i.test(pkg.bin[cli.binName])) {
        import(path.resolve(path.dirname(pkgPath), pkg.bin[cli.binName])).catch(
	    err => {
		console.error(err);
		process.exitCode = 1;
	    }
        );
    } else {
	require(path.resolve(path.dirname(pkgPath), pkg.bin[cli.binName]));
    }
```

**Reply by alexander-akait** [reply]
Can you create reprducible test repo?
